src/test/ui/mismatched_types/cast-rfc0401.stderr

```diff
@@ -20,57 +20,67 @@ error: no field `f` on type `fn() {main}`
 75 |     let _ = main.f as *const u32;
    |                  ^
 
-error: non-scalar cast: `*const u8` as `&u8`
+error[E0605]: non-scalar cast: `*const u8` as `&u8`
```

Comment:
"non-primitive cast"? Primitive is a word people will/should be exposed to when using rust. I don't think "non-scalar" appears much in the docs (though I haven't searched, but I've never seen it).